zebra/zebra_ns.h

```diff
@@ -52,7 +52,12 @@ struct zebra_ns {
 #ifdef HAVE_NETLINK
 	struct nlsock netlink;        /* kernel messages */
 	struct nlsock netlink_cmd;    /* command channel */
-	struct nlsock netlink_dplane; /* dataplane channel */
+
+	/* dplane system's channels: one for outgoing programming,
+	 * for the FIB e.g., and one for incoming events from the OS.
+	 */
+	struct nlsock netlink_dplane_out;
+	struct nlsock netlink_dplane_in;
```

Comment:
is there really need to have 2 sockets ?

Reply:
Yes, like the sockets used by the main pthread: one socket listens for asynchronous incoming events, the other is used to send outbound changes, and it needs to read results synchronously.